flink-format-changelog-json/src/main/java/com/ververica/cdc/formats/json/ChangelogJsonFormatFactory.java

```diff
@@ -108,7 +112,8 @@ public ChangelogMode getChangelogMode() {
             public SerializationSchema<RowData> createRuntimeEncoder(
                     DynamicTableSink.Context context, DataType consumedDataType) {
                 final RowType rowType = (RowType) consumedDataType.getLogicalType();
-                return new ChangelogJsonSerializationSchema(rowType, timestampFormat);
+                return new ChangelogJsonSerializationSchema(
+                        rowType, timestampFormat, encodeDecimalAsPlainNumber);
```

Comment:
For better extension, Can use the `new ChangelogJsonSerializationSchema(rowType, timestampFormat, formatOptions)` ?

Reply:
I modified the constructor at your suggestion, Please review the code again.